uiautomationutilities/perf_tool/readme.md

```diff
@@ -1,155 +1,157 @@
+- [Arguments to the tool](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/paul/update-perf-testing-documentation/uiautomationutilities/perf_tool/readme.md#arguments-to-the-tool)
 
-	- 2. In tag, "Scenarios", add a new scenario with the Measurement ID of the MesurementConfiguration just added in previous sub-step.
-- 4. Push the common code and msal native code to dev branch by raising PRs.
-- 5. Push the C# changes of its repo (i.e. common) into dev branch by raising a PR.
-- 6. Deploy C# tool into the pipeline. (Pipeline information given at the start of this doc). We can deploy C# tool even before merging of PR if need be. As this deployment is done after building at the local machine.
+- [Email report](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/paul/update-perf-testing-documentation/uiautomationutilities/perf_tool/readme.md#email-report)
+
+### Definition of terms:
+[**PerfMarker / CodeMarker:**](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/PerfConstants.java#L36) A marker in code that results in an event to capture timestamp and other details as needed (Memory, Thread ID, CorrelationID etc. and write to file). The data generated from a CodeMarker can be used for Perf, logging, Telemetry etc.
+
+[**Measurement:**](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/uiautomationutilities/perf_tool/PerfDataConfiguration.xml#L4) A pair of markers identified for a transaction. E.g. If can have marker M1 at the beginning of a silent request and another when we successfully received a token say M2. Now M1 and M2 can be paired together if we were interested in measuring time between these. Pairing is an outside concept and has no impact from a code perspective.
+
+[**Scenario:**](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/uiautomationutilities/perf_tool/PerfDataConfiguration.xml#L18-L25) An E2E scenario that we would be interested in measuring. E.g. Silent Token refresh on a 4G network when requesting a token for graph. A scenario can be associated with one or more measurements.
+
+**Base build:** This is a build against which we compare performance of future builds. It would typically be the last release. It is configured on the Perf pipeline using the `baseBuildID` variable
+
+**Basefiles:** These are the code marker files during the run of the base build and help us compare the current run against the base run
+
+**Targetfiles:** These are the code marker files generated during the current run
+
+
+### Steps to add a new marker and see it in the report:
```

Comment:
I think this section could benefit from tighter formatting... example below

## Steps to Add a New Marker

### (Android Common) Declare Constants, Update PerfTool XML, and Update Source as Necessary
1. Lorem ipsum dolor sit amet, consectetur adipiscing elit
2. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
3. etc etc...

### (MSAL/Broker) Define Test Case[s] and Execute
1. Lorem ipsum dolor sit amet, consectetur adipiscing elit
2. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
3. etc etc...

Reply:
Done